### PR TITLE
Update gdb_proxy.py

### PR DESCRIPTION
**Pull request by acidtonic**
Fixed missing check for None on the match.....

Was crashing due to match not having .group(2) on line 45.

After adding simple check for None, everything works for me if I install my fork. Hopefully this can get merged upstream.

If it matters I'm on Gentoo with nvim 0.3.7 and this repo at HEAD.

### CODE REVIEW

**Comment by acidtonic** [comment]
Sorry maybe I'm confused but I never changed anything to a string. First it was int, and I changed it to float assuming whatever code you have downstream uses it numerically.

Let me know if I'm missing something.

File: lib/gdb_proxy.py [hunk 1]
```diff
@@ -42,13 +42,13 @@ def process_info_breakpoints(self, last_src, response):
...
                     match = pattern.fullmatch(fields[-1])   # file.cpp:line
-                    if (match):
+                    if match:
                         is_end_match = last_src.endswith(match.group(1))
                         is_end_match_full_path = \
                             last_src.endswith(os.path.realpath(match.group(1)))
                         if (match and (is_end_match or is_end_match_full_path)):
                             line = match.group(2)
-                            br_id = int(fields[0])
+                            br_id = float(fields[0])
```
[the remark this answers is not in the document]
Comment (acidtonic) [reply lib/gdb_proxy.py]: if a variable = int(something) that varible would be an int, not a string.

[the remark this answers is not in the document]
Comment (sakhnik) [reply lib/gdb_proxy.py]: I meant to assign `br_id` the string `fields[0]` (without int or float), and fix the callers to accept strings instead of ints.

**Comment by sakhnik** [comment]
Sorry, that's my comment was confusing. We realize that `int` wasn't enough, so I suggest go with the string, not float.

**Comment by acidtonic** [comment]
Honestly I'm not sure what you are after.... I've fixed it and it works for
me. I was just hoping to share a simple missed check for an empty
regex upstream.....

My fix was to properly check for non-matching regex matches....

You were the one suggesting I switch to float/int/string since gdb indexes
for breakpoints aren't simply integers. This was a failure of your original
design, not my correction of skipping non-matching regex lines.

I need nothing else other than your code not to crash with calling group()
on a NoneType. (my fix, preventing an immediate crash)
Please implement that and figure out what you want to do with datatypes as
part of another larger design issue. (your overall design fix, correctly
handing breakpoints as either floats or strings.)

**Comment by sakhnik** [comment]
No worries, I can do that. Thanks for your contribution!

**Comment by sakhnik** [comment]
Merged into master manually.

**Pull request closed without merging**